### DISCUSSION TITLE
identify rectangle content (its not a word, image, curve)

### DISCUSSION THREAD

**Discussion by pauljohn32** [jsvine/pdfplumber#641] [Ask for help with specific PDFs]
I'm practicing with pdfplumber to detect fields in US government forms.  Example today is a tax form.  These have lots of checkboxes.  

![check_boxes_empty](https://user-images.githubusercontent.com/344033/162832938-abaae725-9672-4550-b763-5d9016fe3cce.png)

Typically, the pdf are generated by tax prep software.  The checkboxes are not Acrobat form elements, so far in my experience. I attach an example IRS form for testing. 

[f1040.pdf](https://github.com/jsvine/pdfplumber/files/8467674/f1040.pdf)

`.extract_text` and `.extract_words` work as expected for me, except for these checkbox things.  These are not extracted as text or words.   These can be found in the  `rects` for the page.  So I know there's something in there, but what is it?

The visualization tool works, using code below I show highlighted check boxes with ImageMagick tool in pdfplumber. 

![check_boxes](https://user-images.githubusercontent.com/344033/162831550-91bdf9fd-8164-45d4-ad18-b379b95a8367.png)

With the attached `f1040.pdf` file, I run these to make the picture:

```fn1 = "f1040.pdf"
irs =  pdfplumber.open(fn1)
irs1 = irs.pages[0]
irs1rects = irs1.rects
im = irs1.to_image(resolution=150)
irs1checkrects = [j for j in irs1rects if (j['bottom'] > 75) and (j['bottom'] < 77)]
im.draw_rects(irs1checkrects)
```

If I draw the `extract_words` output, the checkboxes are not selected.  That's why I dig through the `rects` to find the coordinates that align.  I hoped that the checkbox itself and the X mark inside it (if there is one) will be some Unicode characters.

I appreciate your advice.

**Comment by jsvine** [comment]
Hi @pauljohn32, and thanks for sharing this interesting example. Although these checkboxes are not Acrobat form elements, they are another kind of interactive interactive element, an "[annotation](https://pspdfkit.com/blog/2018/what-are-annotations/)." These can be accessed via `page.annots`. 

In the [linked](https://notebooksharing.space/view/a09aa83a2ef9ae26df412fa4a2d21c2072962a7075ba7b67a74cfe6cf6e001d2) and attached ([discussion-641.ipynb.txt](https://github.com/jsvine/pdfplumber/files/8476605/discussion-641.ipynb.txt)) notebook, you can see what the data for those annotations look like, and how they change when you've checked the checkboxes (as I have in this file, also referenced in the notebook: [`f1040-checked.pdf`](https://github.com/jsvine/pdfplumber/files/8476619/f1040-checked.pdf)).
